### PR TITLE
Fix bug in Textual Inversion Unloading

### PR DESCRIPTION
**Pull request by bonlime**
# What does this PR do?

Fixes Textual inversion unloading bug. To reproduce:
```python
import torch
from diffusers import AutoPipelineForText2Image

pipeline = AutoPipelineForText2Image.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0")

print(pipeline.text_encoder.text_model.embeddings.token_embedding)
pipeline.load_textual_inversion("sd-concepts-library/gta5-artwork")
pipeline.unload_textual_inversion()
pipeline.load_textual_inversion("sd-concepts-library/gta5-artwork")
pipeline.unload_textual_inversion()
print(pipeline.text_encoder.text_model.embeddings.token_embedding)
```

Would print:
```
Embedding(49408, 768)
Embedding(49409, 768) <- note 49409 instead of 49408 here
```
Due to incorrect tokenizer size, after adding embedding 2nd time, it's not being correctly reset. After this PR the output would be correct:
```
Embedding(49408, 768)
Embedding(49408, 768)
```

## Before submitting
~- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).~
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
~- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.~
~- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).~
~- [ ] Did you write any new necessary tests?~

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by bonlime** [comment]
hi @sayakpaul can you please chech the PR? it's really tiny

**Comment by sayakpaul** [comment]
Thanks! Could you also add / update tests for this?

**Comment by sayakpaul** [comment]
@bonlime a gentle ping here.

**Comment by bonlime** [comment]
@sayakpaul gentle ping worked :)

**Comment by sayakpaul** [comment]
@yiyixuxu could you give this a look?.

File: src/diffusers/loaders/textual_inversion.py [hunk 1]
```diff
@@ -561,6 +561,7 @@ def unload_textual_inversion(
                 tokenizer._added_tokens_encoder[token.content] = last_special_token_id + key_id
                 key_id += 1
         tokenizer._update_trie()
+        tokenizer._update_total_vocab_size()
```
Comment (sayakpaul) [inline src/diffusers/loaders/textual_inversion.py:564]: Would be nice to add a comment explaining this change.

**Review by sayakpaul — COMMENTED** [review]
Just a single comment.

File: src/diffusers/loaders/textual_inversion.py (hunk 1 above)
Comment (bonlime) [reply src/diffusers/loaders/textual_inversion.py:565]: added simple comment for that

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
can you make style?

**Comment by bonlime** [comment]
Fixed style

**Comment by yiyixuxu** [comment]
thanks for the PR!

**Pull request merged**